packages/consumption/src/modules/requests/itemProcessors/readAttribute/ReadAttributeRequestItemProcessor.ts

```diff
@@ -378,7 +386,13 @@ export class ReadAttributeRequestItemProcessor extends GenericRequestItemProcess
             if (responseItem.successorContent instanceof IdentityAttribute) {
                 const { predecessor, successor } = await this.consumptionController.attributes.succeedPeerSharedIdentityAttribute(responseItem.predecessorId, successorParams);
                 return new PeerSharedAttributeSucceededEvent(this.currentIdentityAddress.toString(), predecessor, successor);
-            } else if (responseItem.successorContent.owner === requestInfo.peer) {
+            }
+            const predecessor = await this.consumptionController.attributes.getLocalAttribute(responseItem.predecessorId);
+
+            if (successorParams.shareInfo) {
```

Comment:
Isn't this always true?

Reply:
Yes but not for typescript and without this check TS will not allow us to make the assignment